sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -799,6 +800,77 @@ public CosmosClientBuilder endToEndOperationLatencyPolicyConfig(CosmosEndToEndOp
         return this;
     }
 
+    /**
+     * Sets the {@link CosmosSessionRetryOptions} instance on the client.
+     * <p>
+     * This setting helps in optimizing retry behavior associated with
+     * {@code NOT_FOUND / READ_SESSION_NOT_AVAILABLE} or {@code 404 / 1002} scenarios which happen
+     * when the targeted consistency used by the request is <i>Session Consistency</i> and a
+     * request goes to a region that does not have recent enough data which the
+     * request is looking for.
+     * <p>
+     * DISCLAIMER: Setting {@link CosmosSessionRetryOptions} will modify retry behavior
+     * for all operations or workloads executed through this instance of the client.
+     * <p>
+     * For multi-write accounts:
+     * <ul>
+     *     <li>
+     *         For a read request going to a local read region, it is possible to optimize
+     *         availability by having the request be retried on a remote write region.
+     *     </li>
+     *     <li>
+     *         For a read request going to a local write region, it may or may not help to
+     *         switch to a different write region right away.
+     *     </li>
+     *     <li>
+     *         For a write request going to a local write region, it may or may not help to
+     *         switch to a different write region right away.
+     *     </li>
+     * </ul>
+     * For single-write accounts:
+     * <ul>
+     *     <li>
+     *         If a read request goes to a local read region, it helps to switch to a write region quicker.
+     *     </li>
+     *     <li>
+     *         If a read request goes to a write region, the {@link CosmosSessionRetryOptions} setting does not
```

Comment:
and also if the reads goes to the write region, then we should be able to find the writes as we will loop through all the replicas before the sessionTokenRetryToken kick in

Reply:
Fixed the javadoc in the next iteration.